CONTRIBUTING.rst

```diff
@@ -0,0 +1,662 @@
+IDE integration           - (+) straightforward            - (-) via remote debugging only
+========================= ================================ =====================================
+
+
+Typically, you are recommended to use both of these environments depending on your needs.
+
+Local virtualenv Development Environment
+----------------------------------------
+
+All details about using and running local virtualenv enviroment for Airflow can be found in `LOCAL_VIRTUALENV.rst <LOCAL_VIRTUALENV.rst>`__.
+
+Benefits:
+
+-   Packages are installed locally. No container environment is required.
+
+-   You can benefit from local debugging within your IDE.
+
+-   With the virtualenv in your IDE, you can benefit from autocompletion and running tests directly from the IDE.
+
+Limitations:
+
+-   You have to maintain your dependencies and local environment consistent with
+    other development environments that you have on your local machine.
+
+-   You cannot run tests that require external components, such as mysql,
+    postgres database, hadoop, mongo, cassandra, redis, etc.
+
+    The tests in Airflow are a mixture of unit and integration tests and some of
+    them require these components to be set up. Local virtualenv supports only
+    real unit tests. Technically, to run integration tests, you can configure
+    and install the dependencies on your own, but it is usually complex.
+    Instead, you are recommended to use
+    `Breeze development environment <#breeze-development-environment>`__ with all required packages
+    pre-installed.
+
+-   You need to make sure that your local environment is consistent with other
+    developer environments. This often leads to a "works for me" syndrome. The
+    Breeze container-based solution provides a reproducible environment that is
+    consistent with other developers.
+
+Possible extensions:
+
+-   You are **STRONGLY** encouraged to also install and use `pre-commit hooks <#pre-commit-hooks>`_
+    for your local virtualenv development environment.
+    Pre-commit hooks can speed up your development cycle a lot.
+
+Breeze Development Environment
+------------------------------
+
+All details about using and running Airflow Breeze can be found in
+`BREEZE.rst <BREEZE.rst>`__.
+
+The Airflow Breeze solution is intended to ease your local development as "*It's
+a Breeze to develop Airflow*".
+
+Benefits:
+
+-   Breeze is a complete environment that includes external components, such as
+    mysql database, hadoop, mongo, cassandra, redis, etc., required by some of
+    Airflow tests. Breeze provides a preconfigured Docker Compose environment
+    where all these services are available and can be used by tests
+    automatically.
+
+-   Breeze environment is almost the same as used in `Travis CI <https://travis-ci.com/>`__ automated builds.
+    So, if the tests run in your Breeze environment, they will most likely work in Travis CI as well.
```

Comment:
```suggestion
    So, if the tests run in your Breeze environment, they will work in Travis CI as well.
```
We should be brave, otherwise our emotions will go to other users.